src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -20,45 +20,149 @@ package io.github.yuutoproject.yuutobot.commands
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.utils.data.DataObject
 import net.dv8tion.jda.internal.utils.IOUtil
 import okhttp3.*
 import okhttp3.Callback as OkHttp3Callback
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.RequestBody.Companion.toRequestBody
+import org.slf4j.LoggerFactory
 
-// TODO: get arch to implement the actual command
 class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "does something", "[bg] <char> <text>") {
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+
+    private val backgrounds = listOf(
+        "bath",
+        "beach",
+        "cabin",
+        "camp",
+        "cave",
+        "forest",
+        "messhall"
+    )
+
+    private val characters = listOf(
+        "aiden",
+        "avan",
+        "chiaki",
+        "connor",
+        "eduard",
+        "felix",
+        "goro",
+        "hiro",
+        "hunter",
+        "jirou",
+        "keitaro",
+        "kieran",
+        "knox",
+        "lee",
+        "naoto",
+        "natsumi",
+        "seto",
+        "taiga",
+        "yoichi",
+        "yoshi",
+        "yuri",
+        "yuuto"
+    )
+
+    private val backgroundsString = "`${backgrounds.joinToString("`, `")}`"
+    private val charactersString = "`${characters.joinToString("`, `")}`"
+
+    private val json = "application/json; charset=utf-8".toMediaType()
+
+    // Using ExperimentalStdLib for .removeFirst() in Mutable Lists
+    @ExperimentalStdlibApi
     override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
-        val moveThisToAStorageLocation = OkHttpClient()
+        val now: Long = System.currentTimeMillis()
+
+        if (args.count() < 2) {
+            event.channel.sendMessage("This command requires two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
+            return
+        }
+
+        var character = args.removeFirst().toLowerCase()
+        val background: String
+
+        if (characters.contains(character)) {
+            background = "camp"
+        } else {
+            background = character
+            character = args.removeFirst().toLowerCase()
+        }
+
+        if (!backgrounds.contains(background)) {
+            event.channel.sendMessage("Sorry but I couldn't find `$background` as a location\nAvailable backgrounds are: $backgroundsString").queue()
+            return
+        }
+
+        if (!characters.contains(character)) {
+            event.channel.sendMessage("Sorry, but I don't think that `$character` is a character in Camp Buddy\nAvailable characters are: $charactersString").queue()
+            return
+        }
+
+        if (args.count() < 1) {
+            event.channel.sendMessage("Please provide a message to be written~!").queue()
+            return
+        }
+
+        val text: String = args.joinToString(" ").replace("/[‘’]/g".toRegex(), "'")
+
+        if (text.length > 120) {
```

Comment:
Weren't we going to allow more characters in the string?

Reply:
Lets bump it to 140 for now